Content.Shared/Shipyard/Components/ShipyardListingComponent.cs

```diff
@@ -0,0 +1,18 @@
+using Content.Shared.Shipyard.Prototypes;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype.List;
+
+namespace Content.Shared.Shipyard.Components;
+
+/// <summary>
+///   When applied to a shipyard console, adds all specified shuttles to the list of sold shuttles.
+///   Can also override the name of the console.
+/// </summary>
+[RegisterComponent]
+public sealed partial class ShipyardListingComponent : Component
+{
+    [ViewVariables, DataField(customTypeSerializer: typeof(PrototypeIdListSerializer<VesselPrototype>))]
+    public List<string> Shuttles = new();
+
+    [ViewVariables, DataField("name")]
+    public string? NameOverride = null;
```

Comment:
whats the use case for this, where the entity name itself wouldnt be changed in the prototype as its created, or a mapper wouldnt change the name before saving?

Reply:
It's a redunant thing I think, I made it and then changed things so it was no longer needed... I probably should remove it